### PR TITLE
Use poseidon-py

### LINKED ISSUE #803: Support calculating Poseidon hash
### Feature Request

https://docs.starknet.io/documentation/starknet_versions/upcoming_versions/#poseidon_hash

### PR DESCRIPTION
**Pull request by drknzz**
Closes #803 

- [poseidon-py on pypi](https://pypi.org/project/poseidon-py/0.1.1/)
- [repository on swm-labs](https://github.com/software-mansion-labs/poseidon-py)

### CODE REVIEW

File: starknet_py/hash/sierra_class_hash.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from typing import List
 
-from starkware.cairo.common.poseidon_hash import poseidon_hash_many
+from poseidon_py.poseidon_hash import poseidon_hash_many
```
Comment (war-in) [inline starknet_py/hash/sierra_class_hash.py:3]: ❤️

File: starknet_py/net/models/transaction_test.py [hunk 2]
```diff
@@ -121,7 +121,7 @@ def test_declare_compress_program(balance_contract):
                 signature=[0x1, 0x2],
                 version=2,
             ),
-            1060028572003053454264880444429114465277231377909922229124671748389152379023,
+            669407379128146207315662195646033049577214993241551268214610074015474911988,
```
Comment (war-in) [inline starknet_py/net/models/transaction_test.py:124]: :hmmm: 
any comment on this?

File: starknet_py/tests/e2e/fixtures/contracts.py [hunk 3]
```diff
@@ -32,11 +32,11 @@ def map_compiled_contract() -> str:
 @pytest.fixture(scope="package")
 def sierra_minimal_compiled_contract_and_class_hash() -> Tuple[str, int]:
     """
-    Returns minimal contract compiled to sierra and its class hash.
+    Returns minimal contract compiled to sierra and its compiled class hash.
     """
     return (
         read_contract("precompiled/minimal_contract_compiled.json"),
-        0x56B935FA8AF97EC603A96A3E1870CA193DA4CF08B8DA3F12E023E67A82B5A7E,
+        0x73F17E5E8C771A97CB07BF6024753D514ED9A1B5DE4EC151E06D0926B015694,
```
Comment (war-in) [inline starknet_py/tests/e2e/fixtures/contracts.py:39]: 👀 and this

Comment (drknzz) [reply starknet_py/tests/e2e/fixtures/contracts.py:39]: minimal_contract_compiled has been updated to the latest from cairo repo and so the compiled class hash changed

**Review by war-in — APPROVED** [review]

**Pull request merged**